### PR TITLE
Using defined debug port and not Serial + CC1101 length mode config options

### PR DESCRIPTION
**Pull request by phretor**

### CODE REVIEW

**Comment by phretor** [comment]
Proposal on how to address #54 in the CC1101 module. I can implement and send more pull request for other modules if this approach is acceptable.

File: src/modules/CC1101.cpp [hunk 1]
```diff
@@ -477,7 +480,11 @@ uint8_t CC1101::getLQI() {
 
 size_t CC1101::getPacketLength(bool update) {
   if(!_packetLengthQueried && update) {
-    _packetLength = _mod->SPIreadRegister(CC1101_REG_FIFO);
+
+    uint8_t format = SPIgetRegValue(CC1101_REG_PKTCTRL0, 1, 0);
```
Comment (jgromes) [inline src/modules/CC1101.cpp]: Should be able to use `_packetLengthConfig` variable here instead of SPI read.

File: src/modules/CC1101.cpp [hunk 2]
```diff
@@ -497,6 +504,38 @@ int16_t CC1101::config() {
   return(state);
 }
 
+int16_t CC1101::fixedPacketLengthMode(uint8_t len) {
```
Comment (jgromes) [inline src/modules/CC1101.cpp]: Neither one of the new methods updates `_packetLengthConfig`.

**Comment by phretor** [comment]
Thanks for the thorough review. I'll handle the comments. Should I just make a new pull request or merge new changes with this one?

**Comment by jgromes** [comment]
New PR for the new changes in other radio modules seems more appropriate, thanks.

**Comment by phretor** [comment]
Changeset 112110d should address your requests.

File: src/modules/CC1101.cpp [hunk 3]
```diff
@@ -480,11 +480,12 @@ uint8_t CC1101::getLQI() {
 
 size_t CC1101::getPacketLength(bool update) {
   if(!_packetLengthQueried && update) {
-
-    uint8_t format = SPIgetRegValue(CC1101_REG_PKTCTRL0, 1, 0);
-    if (format == CC1101_LENGTH_CONFIG_VARIABLE) {
+    if (_packetLengthConfig == CC1101_LENGTH_CONFIG_VARIABLE) {
+      _packetLength = _mod->SPIreadRegister(CC1101_REG_FIFO);
+    } else {
       _packetLength = _mod->SPIreadRegister(CC1101_REG_FIFO);
```
Comment (jgromes) [inline src/modules/CC1101.cpp]: I think we should read the length from CC1101_REG_PKTLEN for fixed packets, right?

Comment (phretor) [reply src/modules/CC1101.cpp]: Yeah, sorry. The upcoming one should do. I've started integrating the same changes in other modules, too. Will push in the following hours. For some of them I have hardware that I can test.

**Comment by phretor** [comment]
Commit [a2e64f1](https://github.com/jgromes/RadioLib/pull/55/commits/a2e64f1645a9f6630e7ef2b9ec9abc7e6c335349) should do it.

**Comment by jgromes** [comment]
Looks godd to me - merged, thanks!

**Pull request merged**